core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala

```diff
@@ -96,7 +97,9 @@ abstract class FileCommitProtocol extends Logging {
    * guarantees that files written by different tasks will not conflict.
    */
   @deprecated("use newTaskTempFile(..., spec: FileNameSpec) instead", "3.3.0")
-  def newTaskTempFile(taskContext: TaskAttemptContext, dir: Option[String], ext: String): String
+  def newTaskTempFile(taskContext: TaskAttemptContext, dir: Option[String], ext: String): String = {
+    throw SparkException.mustOverrideOneMethodError("newTaskTempFile")
```

Comment:
In other way, I don't think this is required to achieve your goal.

Reply:
I agree, it is not clear why these changes are being made.
I got confused with the intent as well - though now I realize what is being attempted.